daemons/ipa-slapi-plugins/ipa-extdom-extop/ipa_extdom_common.c

```diff
@@ -97,135 +98,180 @@ static int inc_buffer(size_t buf_max, size_t *_buf_len, char **_buf)
     return 0;
 }
 
-int getpwnam_r_wrapper(size_t buf_max, const char *name,
+int __nss_to_err(enum nss_status errcode)
+{
+    int ret = -1;
+
+    switch(errcode) {
+    case NSS_STATUS_SUCCESS:
+        ret = 0;
+        break;
+    case NSS_STATUS_NOTFOUND:
+        ret = ENOENT;
+        break;
+    case NSS_STATUS_TRYAGAIN:
+        ret = ERANGE;
+        break;
+    case NSS_STATUS_UNAVAIL:
+        ret = ETIMEDOUT;
+        break;
+    }
+
+    return ret;
+}
+
+int getpwnam_r_wrapper(struct ipa_extdom_ctx *ctx, const char *name,
                        struct passwd *pwd, char **_buf, size_t *_buf_len)
 {
     char *buf = NULL;
     size_t buf_len = 0;
-    int ret;
+    int ret, lerrno = 0;
     struct passwd *result = NULL;
+    enum nss_status rc;
 
-    buf = *_buf;
-    buf_len = *_buf_len;
-
-    while (buf != NULL
-            && (ret = getpwnam_r(name, pwd, buf, buf_len, &result)) == ERANGE) {
-        ret = inc_buffer(buf_max, &buf_len, &buf);
-        if (ret != 0) {
-            if (ret == ERANGE) {
-                LOG("Buffer too small, increase ipaExtdomMaxNssBufSize.\n");
+    for (buf = *_buf, buf_len = *_buf_len; buf != NULL;) {
+        rc = back_extdom_getpwnam(ctx->nss_ctx, name, pwd, buf, buf_len, &result, &lerrno);
+        ret = __nss_to_err(rc);
+        if (ret == ERANGE) {
+            ret = inc_buffer(ctx->max_nss_buf_size, &buf_len, &buf);
+            if (ret != 0) {
+                goto done;
             }
-            goto done;
+        } else {
+            break;
```

Comment:
the intention of my previous comment was to remove this else branch, sorry for not being clear

Reply:
it seem like buf != NULL is not really the condition you are looking for in this cycle.
You want to break when ret != ERANGE, then maybe do that ?
The point is to express in the for loop what are the initial conditions and the exit condition, so that a future reader knows exactly what we care for and pay attention to that specific detail and not anciallry variables that just happen to allow us to do a loop:
```
buf = *_buf;
buf_len = *_buf_len;
for (ret = ERANGE; ret != ERANGE;) {
```